builder/pipeline.cc

```diff
@@ -509,6 +597,22 @@ class pipeline_builder {
       body = crop_for_loop(body, f, loop);
       // And make the actual loop.
       body = loop::make(loop.sym(), loop.mode, get_loop_bounds(f, loop), loop.step, body);
+
+      // Wrap loop into crops.
+      for (symbol_id i : input_syms_) {
+        if (!allocation_bounds_[i]) continue;
+        body = crop_buffer::make(i, *allocation_bounds_[i], body);
+      }
+
+      for (int ix = order_.size() - 1; ix >= 0; ix--) {
+        const auto& f = order_[ix];
+
+        for (const auto& o : f->outputs()) {
```

Comment:
I think this should use the real type too. We use `func::input` in a similar loop just above here for example.